packages/plans-grid-next/src/hooks/data-store/use-grid-plans-for-comparison-grid.ts

```diff
@@ -28,12 +44,13 @@ const useGridPlansForComparisonGrid = ( {
 	useCheckPlanAvailabilityForPurchase,
 	useFreeTrialPlanSlugs,
 }: UseGridPlansParams ): GridPlan[] | null => {
+	const comparisonTableIntent = getIntentForComparisonTable( intent ?? 'default' );
```

Comment:
The function already handled `falsy` values. Maybe change the function signature to accept null or such.